arch/arm/lib/memcpymove.h

```diff
@@ -419,43 +419,43 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
         bhi     157f
         bcs     156f
         bmi     155f
-        memcpy_long_inner_loop  backwards, 0
-155:    memcpy_long_inner_loop  backwards, 1
-156:    memcpy_long_inner_loop  backwards, 2
-157:    memcpy_long_inner_loop  backwards, 3
+        memcpy_long_inner_loop  \backwards, 0
+155:    memcpy_long_inner_loop  \backwards, 1
+156:    memcpy_long_inner_loop  \backwards, 2
+157:    memcpy_long_inner_loop  \backwards, 3
 
         UNWIND( .fnend )
 
         UNWIND( .fnstart )
         UNWIND( .save {D, DAT1, DAT2, lr} )
 
 160:    /* Medium case */
-        preload_all  backwards, 0, 0, S, N, DAT2, OFF
+        preload_all  \backwards, 0, 0, S, N, DAT2, OFF
         sub     N, N, #16     /* simplifies inner loop termination */
- .if backwards
+ .if \backwards
         ands    DAT2, D, #15
         beq     164f
  .else
         ands    DAT2, D, #15
         beq     164f
         rsb     DAT2, DAT2, #16
  .endif
-        memcpy_leading_15bytes backwards, align
+        memcpy_leading_15bytes \backwards, 1
```

Comment:
Should this be \align ?

Reply:
I want to know why, too 😂

**The test assembly source for a quick reproduction**
```
    .text
    .arch armv6
    .object_arch armv4
    .arm
    .p2align 2

.macro should_not_assemble_macro arg
    .if \arg == 0
        add       a1, a2, #4
    .else
        sub       a1, a3, #2
    .endif
.endm

should_not_assemble:
    should_not_assemble_macro arg

    .type should_not_assemble, %function

    .end
```

**GCC**
```
(raspbian-bookworm-armhf)a130ft@berget:~/work/should-not-assemble$ gcc -o should-not-assemble.o -c should-not-assemble.S 
(raspbian-bookworm-armhf)a130ft@berget:~/work/should-not-assemble$ objdump --disassemble should-not-assemble.o

should-not-assemble.o:     file format elf32-littlearm


Disassembly of section .text:

00000000 <should_not_assemble>:
   0:	e2420002 	sub	r0, r2, #2
(raspbian-bookworm-armhf)a130ft@berget:~/work/should-not-assemble$ 
```

**LLVM (clang)**
```
(raspbian-bookworm-armhf)a130ft@berget:~/work/should-not-assemble$ clang -o should-not-assemble.o -c should-not-assemble.S 
<instantiation>:1:5: error: expected absolute expression
.if arg == 0
    ^
should-not-assemble.S:16:5: note: while in macro instantiation
    should_not_assemble_macro arg
    ^
(raspbian-bookworm-armhf)a130ft@berget:~/work/should-not-assemble$ 
```

So gcc insists that `arg == 0` expended out of `\arg == 0` has the acceptable syntax and evaluates to false.